sim/core/focus.go

```diff
@@ -29,38 +29,35 @@ type focusBar struct {
 	cumulativeFocusDecisionThresholds []int
 
 	nextFocusTick time.Duration
-	isPlayer bool
+	isPlayer      bool
 
 	// Multiplies focus regen from ticks.
 	FocusTickMultiplier float64
 
-	regenMetrics        *ResourceMetrics
-	FocusRefundMetrics *ResourceMetrics
+	regenMetrics       *ResourceMetrics
+	focusRefundMetrics *ResourceMetrics
 }
 
 func (unit *Unit) EnableFocusBar(maxFocus float64, baseFocusPerSecond float64, isPlayer bool) {
 	unit.SetCurrentPowerBar(FocusBar)
 
 	unit.focusBar = focusBar{
-		unit:                 unit,
+		unit:                unit,
 		maxFocus:            max(100, maxFocus),
 		FocusTickMultiplier: 1,
-		isPlayer: isPlayer,
-		baseFocusPerSecond: baseFocusPerSecond,
-		regenMetrics:         unit.NewFocusMetrics(ActionID{OtherID: proto.OtherAction_OtherActionFocusRegen}),
-		FocusRefundMetrics:  unit.NewFocusMetrics(ActionID{OtherID: proto.OtherAction_OtherActionRefund}),
+		isPlayer:            isPlayer,
+		baseFocusPerSecond:  baseFocusPerSecond,
+		regenMetrics:        unit.NewFocusMetrics(ActionID{OtherID: proto.OtherAction_OtherActionFocusRegen}),
+		focusRefundMetrics:  unit.NewFocusMetrics(ActionID{OtherID: proto.OtherAction_OtherActionRefund}),
 	}
 }
 
 // Computes the focus thresholds.
-func (fb *focusBar) setupFocusThresholds() {
-	if fb.unit == nil {
-		return
-	}
+func (unit *Unit) SetupFocusThresholds() {
```

Comment:
why is this a Unit function now? To me it made sense more to be a function of the focus bar as you need the focus bar for this function to do anything

Reply:
Because I couldn't access the focusBar anywhere outside, except where it was. Maybe I was wrong, but whatever go autocomplete thought `hunter.unit.focusBar` was not a thing I could do.

Reply:
Ahh yeah that is true hmm. Now i'm conflicted about the first review I did. Do we want to keep the setup of the thresholds in the core so we dont have to expose the focusBar but then we have a bit of class code in the core. Which of the 2 is the lesser evil. Right now you have a method that you can call on any unit that works on focus bar but not every unit has a focus bar. Maybe its better if you go back to seting up the thresholds from core.

Reply:
Aight, reverted